chunks 1,5,6,11: Set base paragraph indentation 5n (upstream commit 5d2e49f8).
chunk 2: Tweak section headers (local change).
chunk 3: For .TH "", set the title to "UNTITLED" (to be pushed upstream)
chunk 4: Set the default operating system name (local change).
chunk 7: Fix .HP vertical spacing regression (upstream f607ea44, d483834a).
chunk 8-10: Fix .ME and .UE tail argument regression (to be pushed upstream).

Index: tmac/an.tmac
--- tmac/an.tmac.orig
+++ tmac/an.tmac
@@ -170,7 +170,7 @@
 .
 .de an-reset-margin-and-inset-level
 .  nr an-inset-level 1
-.  nr an-margin \\n[IN]
+.  nr an-margin \\n[BP]
 .  nr an-saved-margin1 \\n[IN]
 .  nr an-prevailing-indent \\n[IN]
 .  nr an-saved-prevailing-indent1 \\n[IN]
@@ -195,7 +195,7 @@
 .  ds an*section1 General Commands Manual\"
 .  ds an*section2 System Calls Manual\"
 .  ds an*section3 Library Functions Manual\"
-.  ds an*section4 Kernel Interfaces Manual\"
+.  ds an*section4 Device Drivers Manual\"
 .  ds an*section5 File Formats Manual\"
 .  ds an*section6 Games Manual\"
 .  ds an*section7 Miscellaneous Information Manual\"
@@ -285,7 +285,8 @@
 .  nr an-is-in-diversion 0
 .  nr an*is-in-example 0
 .
-.  ds an*topic "\\$1\"
+.  ie '\\$1'' .ds an*topic "UNTITLED\"
+.  el         .ds an*topic "\\$1\"
 .  if \\n[CT] .stringup an*topic
 .  ds an*section "\\$2\"
 .  ie (\\n[.$] > 4) .ds an-extra3 "\\$5\"
@@ -306,7 +307,7 @@
 .
 .  ds an-extra1 "\\$3\"
 .  ie (\\n[.$] > 3)    .ds an-extra2 "\\$4\"
-.  el                  .ds an-extra2 \" empty; but .AT/.UC can override
+.  el                  .ds an-extra2 "OpenBSD\"
 .
 .  if '\\*[an-extra1]'' \{\
 .    ds an-msg .\\$0 missing third argument; suggest document\"
@@ -681,7 +682,7 @@ contains unsupported escape sequence
 .  an-break-paragraph
 .  an-reset-margin-and-inset-level
 .  fi
-.  in \\n[an-margin]u
+.  in \\n[BP]u
 .  ti 0
 .  nr an-devtag-needs-end-of-heading 1
 .  DEVTAG-SH 1
@@ -708,7 +709,7 @@ contains unsupported escape sequence
 .  an-break-paragraph
 .  an-reset-margin-and-inset-level
 .  fi
-.  in \\n[IN]u
+.  in \\n[BP]u
 .  ti \\n[SN]u
 .  nr an-devtag-needs-end-of-heading 1
 .  DEVTAG-SH 2
@@ -805,7 +806,6 @@ contains unsupported escape sequence
 .\" Set a paragraph with a hanging indentation.
 .\" .HP [indent]
 .de1 HP
-.  if !\\n[mS] \\*[an-deprecation-warn]\c
 .  an-break-paragraph
 .  ne (1v + 1u)
 .  if \\n[.$] .nr an-prevailing-indent (n;\\$1)
@@ -1112,13 +1112,13 @@ contains unsupported escape sequence
 .        nop \X^tty: link \\*[an*prefix]\\*[an*hyperlink]^\
 \\*[an*hyperlink]\X^tty: link^\c
 .    \}
-.    nop \&\\$1\"
+.    nop \&\\$*\"
 .  \}
 .  \" If not hyperlinking, format URI in angle brackets.  There was no
 .  \" diversion, so the link text has already been formatted normally.
 .  el \{\
 .    nh
-.    nop \\[la]\\*[an*hyperlink]\\[ra]\\$1
+.    nop \\[la]\\*[an*hyperlink]\\[ra]\\$*
 .    hy \\n[an*hyphenation-mode]
 .  \}
 .
@@ -1139,7 +1139,7 @@ contains unsupported escape sequence
 .\" punctuation (or, rarely, other text) after link text.
 .\" .ME [trailing-text]
 .de1 ME
-.  an*end-hyperlink \\$1
+.  an*end-hyperlink \\$*
 .  rm an*prefix
 ..
 .
@@ -1157,7 +1157,7 @@ contains unsupported escape sequence
 .\" punctuation (or, rarely, other text) after link text.
 .\" .UE [trailing-text]
 .de1 UE
-.  an*end-hyperlink \\$1
+.  an*end-hyperlink \\$*
 .  rm an*prefix
 ..
 .
@@ -1323,6 +1323,10 @@ contains unsupported escape sequence
 .  if !r PDFOUTLINE.FOLDLEVEL .nr PDFOUTLINE.FOLDLEVEL 1
 .  if !r PDFHREF.VIEW.LEADING .nr PDFHREF.VIEW.LEADING 10p
 .\}
+.
+.\" base paragraph indentation
+.if !r BP \
+.  nr BP 5n
 .
 .\" continuous rendering (one long page)
 .if !r cR \{\
